src/platform/i386/lapic.c

```diff
@@ -86,26 +87,35 @@ lapic_read_reg(u32_t off)
 
 void
 lapic_set_timer(int timer_type, cycles_t deadline)
-{	
-	u32_t high, low;
-	int retries = RETRY_ITERS;
+{
+	u64_t now;
 
-	if (timer_type != TSC_DEADLINE) {
-		printk("Mode not supported\n");
-		assert(0);
-	}
+	assert (timer_type != PERIODIC);
```

Comment:
The problem with putting the assertion here rather than structuring your conditionals like below is 

1. future changes that add more modes will not trigger an assertion, but won't hit one of the conditions.
2. it isn't really documenting the code well.  "we don't do periodic" is not the same as "we do oneshot and deadline"